### PR TITLE
Update book for pest 2.0

### PR DESCRIPTION
**Pull request by wirelyre**
Definitely could use a quick proofreading to make sure I didn't miss anything!

The second commit in this PR breaks every link to the API docs. I can split it off into another PR if we want to wait until the release to make sure they'll work.

### CODE REVIEW

**Comment by dragostis** [comment]
We can easily merge this and simply update the book once we're ready to release.

**Comment by wirelyre** [comment]
Thanks what I was thinking. Does it look like I found all the changes in the text? I copy-pasted the examples in the code blocks and they all compile.

**Comment by wirelyre** [comment]
Great, thanks for those changes.

I decided to break the built-in rules into their own section since there are so many of them. Is that the right place, or should we document them in the `pest` crate?

@CAD97, do you want to review [the Unicode section](https://github.com/pest-parser/book/blob/7cc5d7413ed1d92ca2361799e4a693c9b117e453/src/grammars/built-ins.md#unicode-rules)? I'd love to know if I wrote anything misleading or wrong.

File: src/grammars/built-ins.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+[The Unicode Standard]: https://www.unicode.org/versions/latest/
+
+### General categories
+
+Formally, categories are non-overlapping: each Unicode character belongs to
+exactly one category, and no category contains another. However, since certain
+groups of categories are often useful together, `pest` exposes the hierarchy of
+categories below. For example, the rule `CASED_LETTER` is not technically a
+Unicode general category; it instead matches characters that are
+`UPPERCASE_LETTER` or `LOWERCASE_LETTER`, which *are* general categories.
```
Comment (dragostis) [inline src/grammars/built-ins.md]: This is a super nice touch.

**Comment by CAD97** [comment]
The Unicode section looks great!

Though now of course I am wondering if you could implement the segmentation algorithm in pest 😅

**Pull request merged**